index.js

```diff
@@ -0,0 +1,203 @@
+  }
+}
+
+
+
+const newIngredientForm = document.getElementById("new-ingredient-form");
+if (newIngredientForm) {
+  newIngredientForm.addEventListener("submit", (event) => {
+    event.preventDefault();
+
+    const newIngredientName = document.getElementById("ingredient-name").value;
+    const newIngredientAmount =
+      document.getElementById("ingredient-amount").value;
+
+    if (!newIngredientName) {
+      alert("Please provide ingredient name.");
+      return;
+    }
+
+    const currentRecipe = recipeObject[recipeObject.length - 1];
+    currentRecipe.ingredients.push({
+      NAME: newIngredientName,
+      AMOUNT: newIngredientAmount,
+    });
+
+    document.getElementById("ingredient-name").value = "";
+    document.getElementById("ingredient-amount").value = "";
+
+    setIngredients(currentRecipe);
+  });
+} else {
+  console.error("Element #new-recipe-form not found");
+}
+
+
+
+
+function addNewRecipe(newRecipeData) {
+  const newRecipe = {
+    id: recipeObject.length + 1,
+    title: newRecipeData.title,
+    picture_url: newRecipeData.imageUrl,
+    description: newRecipeData.description,
+    ingredients: newRecipeData.ingredients,
+    preparationSteps: newRecipeData.preparationSteps,
+  };
+
+  recipeObject.push(newRecipe);
+
+  setRecipe(newRecipe);
+}
+
+const newRecipeForm = document.getElementById("new-recipe-form");
+if (newRecipeForm) {
+  newRecipeForm.addEventListener("submit", (event) => {
+    event.preventDefault();
+
+    const title = document.getElementById("title").value;
+    const imageUrl = document.getElementById("image-url").value;
+    const description = document.getElementById("description").value;
+    const ingredientsInput = document
+      .getElementById("ingredients")
+      .value.split("\n");
+    const preparationSteps = document
+      .getElementById("preparation-steps")
+      .value.split("\n");
+
+    if (ingredientsInput.length < 5) {
+      alert("Please provide at least 5 ingredients.");
+      return;
+    }
+
+    const ingredients = [];
+
+    ingredientsInput.forEach((ingredientLine) => {
+      const parts = ingredientLine.split(":");
+
+      if (parts.length !== 2) {
+        alert(
+          "Invalid ingredient format. Please use 'Ingredient Name: Amount' format."
+        );
+        return;
+      }
+
+      const name = parts[0].trim();
+      const amount = parts[1].trim();
+
+      ingredients.push({ NAME: name, AMOUNT: amount });
+    });
+
+    const newRecipeData = {
+      id: recipeObject.length + 1,
+      title,
+      imageUrl,
+      description,
+      ingredients,
+      preparationSteps,
+    };
+    console.log(recipeObject);
+    addNewRecipe(newRecipeData);
+    document.getElementById("title").value = "";
+    document.getElementById("image-url").value = "";
+    document.getElementById("description").value = "";
+    document.getElementById("ingredients").value = "";
+    document.getElementById("preparation-steps").value = "";
+  });
```

Comment:
A user is complaining that they cannot seem to add their favorite peanut butter cookie recipe. Can you figure out why??

=======================================

Peanut Butter Cookies
Ingredients:

1 cup peanut butter (smooth or chunky)
1 cup granulated sugar
1 large egg
Instructions:

Preheat your oven to 350°F (175°C).
In a mixing bowl, combine the peanut butter, sugar, and egg. Mix until the ingredients are well combined and form a dough.
Roll the dough into small balls (about 1 inch in diameter) and place them on a baking sheet lined with parchment paper.
Use a fork to flatten each ball slightly and create a crisscross pattern on top.
Bake in the preheated oven for 10-12 minutes, or until the edges are golden.
Allow the cookies to cool on the baking sheet for a few minutes before transferring them to a wire rack to cool completely.

===================================

Reply:
The problem could happen in the ingredients part. Cause the [project-description](https://github.com/HackYourFuture-CPH/js-projects/blob/main/projects/recipe-app/recipe-app.md) asked me to "Implement a form to add a new recipe with a minimum of 5 ingredients." So It will not receive the Peanut Butter Cookies recipe since it has less than 5 ingredients. And the text u provide is not in the required ingredient-input-format, which is "IngredientName: Amount". For example it supposed to be like "peanut butter : 1 cup".

Reply:
Great catch. You are completely right, and I should have read the requirements more carefully 😇 . 

Often times, a lot of issues you will face while working on an application are not necessarily due to a bug in the code but rather because the system was designed under wrong assumptions. In this case "we" assumed that a recipe has at least 5 ingredients, which we can see may not always be the case. 

You will always have to make some assumptions when developing an application, but it can sometimes be very hard. Luckily, us developers rarely have to do that. 
Here is an example about names 😉 
https://www.kalzumeus.com/2010/06/17/falsehoods-programmers-believe-about-names/

Reply:
quit sjovt 🤣 just like an invisible battle between human ingenuity and computer logic, neither side wins ever, but just leaves us alone scratching our head.🙉